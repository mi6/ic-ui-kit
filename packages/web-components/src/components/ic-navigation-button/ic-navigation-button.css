@import "../../global/normalize.css";

:host {
  display: block;
  margin-right: var(--ic-space-xs);
  list-style: none;
}

:host(.in-side-menu) {
  margin-right: 0;
}

:host::part(button) {
  height: 100%;
}
