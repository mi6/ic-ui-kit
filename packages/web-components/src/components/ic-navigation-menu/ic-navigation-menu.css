/**
 * @prop --ic-z-index-navigation-menu: z-index of navigation menu
 */

:host {
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  min-height: 100% !important;
  background: rgb(0 0 0 / 60%);
  z-index: var(--ic-z-index-navigation-menu);
}

.popout-modal {
  position: absolute;
  inset: 0;
}

.popout-menu {
  position: fixed;
  top: 0;
  right: 0;
  width: 16rem;
  bottom: 0;
  background-color: var(--ic-architectural-20);
  color: var(--ic-color-primary-text);
  box-shadow: var(--ic-elevation-overlay);
  overflow-y: auto;
  overflow-x: hidden;
}

:host(.inline) .popout-menu {
  position: absolute;
}

.menu-close-button-container {
  position: relative;
  left: 11.875rem;
  padding: var(--ic-space-md) 0;
}

.nav-group-first {
  padding-bottom: 0;
}

.menu-buttons-container {
  padding: var(--ic-space-sm) 0;
  border-bottom: var(--ic-space-1px) solid var(--ic-architectural-200);
}

.menu-buttons-container-nav-item-above {
  margin-top: var(--ic-space-md);
  border-top: var(--ic-space-1px) solid var(--ic-architectural-200);
}

.menu-status-version-container {
  display: flex;
  flex-wrap: wrap;
  gap: var(--ic-space-xs);
  margin: var(--ic-space-md) var(--ic-space-md) var(--ic-space-xl);
}

.menu-status {
  background-color: var(--ic-architectural-500);
  color: var(--ic-color-white-text);
  border-radius: var(--ic-space-md);
  width: fit-content;
  padding: var(--ic-space-xxs) var(--ic-space-sm);
}

.menu-version {
  border-radius: var(--ic-space-md);
  background-color: var(--ic-architectural-100);
  padding: var(--ic-space-xxs) var(--ic-space-sm);
}

.menu-status-text,
.menu-version-text {
  overflow-wrap: break-word;
}

.navigation-landmark-text {
  position: absolute;
  width: var(--ic-space-1px);
  height: var(--ic-space-1px);
  padding: 0;
  margin: calc(-1 * var(--ic-space-1px));
  overflow: hidden;
}
