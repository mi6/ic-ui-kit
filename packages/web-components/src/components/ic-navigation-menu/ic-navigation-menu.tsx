import {
  Component,
  Element,
  Host,
  Prop,
  h,
  Event,
  EventEmitter,
  Listen,
} from "@stencil/core";

import { getSlot, getSlotContent, isSlotUsed } from "../../utils/helpers";

/**
 * @slot navigation - Content will be rendered at top of panel.
 * @slot buttons - Content will be rendered above version info and below navigation.
 */
@Component({
  tag: "ic-navigation-menu",
  styleUrl: "ic-navigation-menu.css",
  shadow: true,
})
export class NavigationMenu {
  private closeButton: HTMLIcButtonElement = null;
  private hasButtons: boolean = false;
  private hasNavigation: boolean = false;
  private lastTabStop: HTMLElement = null;
  private navBarEl: HTMLIcTopNavigationElement;
  private navGroupFirst: boolean = false;
  private navItemAboveButtons: boolean = false;

  @Element() el: HTMLIcNavigationMenuElement;

  /**
   * The status info to display.
   */
  @Prop() status: string = "";

  /**
   * The version info to display.
   */
  @Prop() version: string = "";

  /**
   * @internal - Emitted when the menu is closed.
   */
  @Event() icNavigationMenuClose: EventEmitter<void>;

  componentWillLoad(): void {
    this.navBarEl = document.querySelector("ic-top-navigation");
    const navigationEls = getSlotContent(this.el, "navigation");
    if (navigationEls) {
      this.hasNavigation = true;
      const lastEl = navigationEls[navigationEls.length - 1] as HTMLElement;
      if (lastEl.tagName === "IC-NAVIGATION-ITEM") {
        this.navItemAboveButtons = true;
      }
      const firstEl = navigationEls[0] as HTMLElement;
      if (firstEl.tagName === "IC-NAVIGATION-GROUP") {
        this.navGroupFirst = true;
      }
    }
    const buttonContent = getSlotContent(this.el, "buttons");

    if (buttonContent !== null) {
      this.hasButtons = true;
      this.lastTabStop = buttonContent[buttonContent.length - 1] as HTMLElement;
    } else {
      if (this.hasNavigation) {
        const lastEl = navigationEls[navigationEls.length - 1] as HTMLElement;
        //check for slotted content i.e. react router link
        const slotContent = getSlot(lastEl, "navigation-item");
        if (slotContent !== null) {
          this.lastTabStop = slotContent as HTMLElement;
        } else {
          this.lastTabStop = lastEl;
        }
      }
    }
  }

  componentDidLoad(): void {
    this.focusCloseButton();
  }

  componentWillRender(): void {
    this.hasNavigation = isSlotUsed(this.el, "navigation");
    this.hasButtons = isSlotUsed(this.el, "buttons");
  }

  @Listen("navItemClicked")
  navItemClickHandler(): void {
    this.closeMenu();
  }

  @Listen("keydown", { target: "document" })
  handleKeyDown(ev: KeyboardEvent): void {
    if (ev.key === "Tab") {
      if (ev.shiftKey) {
        if (document.activeElement === this.navBarEl) {
          ev.preventDefault();
          this.focusLastTabStop();
        }
      } else if (
        document.activeElement === this.lastTabStop ||
        this.lastTabStop === null
      ) {
        ev.preventDefault();
        this.focusCloseButton();
      }
    } else if (ev.key === "Escape") {
      this.closeMenu();
    }
  }

  private closeMenu = () => {
    this.icNavigationMenuClose.emit();
  };

  private focusCloseButton = () => {
    if (this.closeButton.focus) {
      this.closeButton.focus();
    }
  };

  private focusLastTabStop = () => {
    let focusEl;
    if (this.lastTabStop !== null) {
      switch (this.lastTabStop.tagName) {
        case "IC-NAVIGATION-BUTTON":
        case "IC-NAVIGATION-ITEM":
        case "IC-NAVIGATION-GROUP":
          focusEl = this.lastTabStop as HTMLElement;
          focusEl.focus();
          break;
        case "A":
          this.lastTabStop.focus();
          break;
      }
    }
  };

  render() {
    return (
      <Host>
        <div class="popout-modal" onClick={this.closeMenu}></div>
        <div
          class="popout-menu"
          role="dialog"
          aria-modal="true"
          aria-label={`${this.hasNavigation ? "Navigation" : "App"} menu`}
        >
          <span
            aria-hidden="true"
            id="navigation-landmark-text"
            class="navigation-landmark-text"
          >
            Main navigation
          </span>
          <nav
            aria-labelledby="navigation-landmark-text"
            aria-hidden={this.hasNavigation ? "false" : "true"}
          >
            <div
              class={{
                ["menu-close-button-container"]: true,
                ["nav-group-first"]: this.navGroupFirst,
              }}
            >
              <ic-button
                ref={(el) => (this.closeButton = el)}
                id="menu-close-button"
                class="menu-close-button"
                variant="icon"
                size="large"
                aria-label={`Close ${
                  this.hasNavigation ? "navigation" : "app"
                } menu`}
                onClick={this.closeMenu}
              >
                <svg
                  slot="icon"
                  width="14"
                  height="14"
                  viewBox="0 0 14 14"
                  fill="none"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z"
                    fill="#1759BC"
                  />
                </svg>
              </ic-button>
            </div>
            {this.hasNavigation && <slot name="navigation"></slot>}
          </nav>
          {this.hasButtons && (
            <div
              class={{
                ["menu-buttons-container"]: true,
                ["menu-buttons-container-nav-item-above"]:
                  this.navItemAboveButtons,
              }}
            >
              <slot name="buttons"></slot>
            </div>
          )}
          {(this.status !== "" || this.version !== "") && (
            <div class="menu-status-version-container">
              {this.status !== "" && (
                <div class="menu-status">
                  <ic-typography
                    variant="label-uppercase"
                    aria-label="app tag"
                    class="menu-status-text"
                  >
                    {this.status}
                  </ic-typography>
                </div>
              )}
              {this.version !== "" && (
                <div class="menu-version">
                  <ic-typography
                    variant="label"
                    class="menu-version-text"
                    aria-label="app version"
                  >
                    {this.version}
                  </ic-typography>
                </div>
              )}
            </div>
          )}
        </div>
      </Host>
    );
  }
}
