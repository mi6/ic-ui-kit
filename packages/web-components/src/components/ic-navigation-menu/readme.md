# ic-navigation-menu



<!-- Auto Generated Below -->


## Properties

| Property  | Attribute | Description                  | Type     | Default |
| --------- | --------- | ---------------------------- | -------- | ------- |
| `status`  | `status`  | The status info to display.  | `string` | `""`    |
| `version` | `version` | The version info to display. | `string` | `""`    |


## Slots

| Slot           | Description                                                       |
| -------------- | ----------------------------------------------------------------- |
| `"buttons"`    | Content will be rendered above version info and below navigation. |
| `"navigation"` | Content will be rendered at top of panel.                         |


## CSS Custom Properties

| Name                           | Description                |
| ------------------------------ | -------------------------- |
| `--ic-z-index-navigation-menu` | z-index of navigation menu |


## Dependencies

### Used by

 - [ic-top-navigation](../ic-top-navigation)

### Depends on

- [ic-button](../ic-button)
- [ic-typography](../ic-typography)

### Graph
```mermaid
graph TD;
  ic-navigation-menu --> ic-button
  ic-navigation-menu --> ic-typography
  ic-button --> ic-loading-indicator
  ic-button --> ic-tooltip
  ic-loading-indicator --> ic-typography
  ic-tooltip --> ic-typography
  ic-top-navigation --> ic-navigation-menu
  style ic-navigation-menu fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


