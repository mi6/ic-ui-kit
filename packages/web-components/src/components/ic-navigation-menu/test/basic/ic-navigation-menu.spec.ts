import { newSpecPage } from "@stencil/core/testing";
import { Button } from "../../../ic-button/ic-button";
import { NavigationGroup } from "../../../ic-navigation-group/ic-navigation-group";
import { NavigationItem } from "../../../ic-navigation-item/ic-navigation-item";
import { NavigationMenu } from "../../ic-navigation-menu";
import { NavigationButton } from "../../../ic-navigation-button/ic-navigation-button";
import { waitForNavGroupLoad } from "../../../../testspec.setup";

describe("ic-navigation-menu", () => {
  it("should render with version and status", async () => {
    const page = await newSpecPage({
      components: [NavigationMenu],
      html: `<ic-navigation-menu version="v1.0.0" status="Beta"></ic-navigation-menu>`,
    });
    expect(page.root).toMatchSnapshot("renders-with-version-and-status");
  });

  it("should test menu close emits event", async () => {
    const page = await newSpecPage({
      components: [NavigationMenu],
      html: `<ic-navigation-menu version="v1.0.0" status="Beta"></ic-navigation-menu>`,
    });

    const eventSpy = jest.fn();
    page.win.addEventListener("icNavigationMenuClose", eventSpy);
    await page.rootInstance.navItemClickHandler();
    await page.waitForChanges();
    expect(eventSpy).toHaveBeenCalled();
  });

  it("should test key down handler", async () => {
    Object.defineProperty(document, "activeElement", { value: null });

    const page = await newSpecPage({
      components: [NavigationMenu],
      html: `<ic-navigation-menu version="v1.0.0" status="Beta"></ic-navigation-menu>`,
    });

    const eventSpy = jest.fn();
    page.win.addEventListener("icNavigationMenuClose", eventSpy);
    await page.rootInstance.handleKeyDown({ key: "Escape" });
    await page.waitForChanges();
    expect(eventSpy).toHaveBeenCalled();

    await page.rootInstance.handleKeyDown({
      key: "Tab",
      preventDefault: (): void => null,
    });
    await page.waitForChanges();

    await page.rootInstance.handleKeyDown({
      key: "Tab",
      shiftKey: true,
      preventDefault: (): void => null,
    });
    await page.waitForChanges();
  });

  it("should test slotted buttons", async () => {
    const page = await newSpecPage({
      components: [NavigationMenu, NavigationButton, Button],
      html: `<ic-navigation-menu version="v1.0.0" status="Beta">
        <ic-navigation-button
          slot="buttons"
          label="button1"
          onclick="alert('test')"
        >
          <svg
            slot="icon"
            xmlns="http://www.w3.org/2000/svg"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            fill="#000000"
          >
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
            />
          </svg>
        </ic-navigation-button>
      </ic-navigation-menu>`,
    });

    //trigger focus
    await page.rootInstance.handleKeyDown({
      key: "Tab",
      shiftKey: true,
      preventDefault: (): void => null,
    });
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot("renders-with-slotted-buttons");
  });

  it("should test slotted navigation item", async () => {
    const page = await newSpecPage({
      components: [NavigationMenu, NavigationItem],
      html: `<ic-navigation-menu version="v1.0.0" status="Beta">
        <ic-navigation-item slot="navigation" label="Navigation 1" href="/"></ic-navigation-item>
      </ic-navigation-menu>`,
    });

    //trigger focus
    await page.rootInstance.handleKeyDown({
      key: "Tab",
      shiftKey: true,
      preventDefault: (): void => null,
    });
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot("renders-with-slotted-navigation-item");
  });

  it("should test slotted navigation group", async () => {
    const page = await newSpecPage({
      components: [NavigationMenu, NavigationGroup],
      html: `<ic-navigation-menu version="v1.0.0" status="Beta">
        <ic-navigation-group
          slot="navigation"
          label="Navigation group"
          expandable="true"
        >
        </ic-navigation-group>
      </ic-navigation-menu>`,
    });
    await waitForNavGroupLoad();

    //trigger focus
    await page.rootInstance.handleKeyDown({
      key: "Tab",
      shiftKey: true,
      preventDefault: (): void => null,
    });
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot("renders-with-slotted-navigation-group");
  });

  it("should test slotted navigation and buttons", async () => {
    const page = await newSpecPage({
      components: [NavigationMenu, NavigationItem, NavigationButton, Button],
      html: `<ic-navigation-menu version="v1.0.0" status="Beta">
        <ic-navigation-button
          label="button1"
          slot="buttons"
          onclick="alert('test')"
        >
          <svg
            slot="icon"
            xmlns="http://www.w3.org/2000/svg"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            fill="#000000"
          >
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
            />
          </svg>
        </ic-navigation-button>
        <ic-navigation-item slot="navigation" label="Navigation 1" href="/"></ic-navigation-item>
      </ic-navigation-menu>`,
    });

    //trigger focus
    await page.rootInstance.handleKeyDown({
      key: "Tab",
      shiftKey: true,
      preventDefault: (): void => null,
    });
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot(
      "renders-with-slotted-navigation-and-buttons"
    );
  });

  it("should test slotted link", async () => {
    const page = await newSpecPage({
      components: [NavigationMenu],
      html: `<ic-navigation-menu version="v1.0.0" status="Beta">
        <a href="/"
          slot="navigation"
        >
      </ic-navigation-menu>`,
    });

    //trigger focus
    await page.rootInstance.handleKeyDown({
      key: "Tab",
      shiftKey: true,
      preventDefault: (): void => null,
    });
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot("renders-with-slotted-link");
  });
});
