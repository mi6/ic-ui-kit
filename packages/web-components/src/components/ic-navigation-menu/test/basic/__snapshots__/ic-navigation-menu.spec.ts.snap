// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-navigation-menu should render with version and status: renders-with-version-and-status 1`] = `
<ic-navigation-menu status="Beta" version="v1.0.0">
  <mock:shadow-root>
    <div class="popout-modal"></div>
    <div aria-label="App menu" aria-modal="true" class="popout-menu" role="dialog">
      <span aria-hidden="true" class="navigation-landmark-text" id="navigation-landmark-text">
        Main navigation
      </span>
      <nav aria-hidden="true" aria-labelledby="navigation-landmark-text">
        <div class="menu-close-button-container">
          <ic-button aria-label="Close app menu" class="menu-close-button" id="menu-close-button" size="large" variant="icon">
            <svg fill="none" height="14" slot="icon" viewBox="0 0 14 14" width="14" xmlns="http://www.w3.org/2000/svg">
              <path d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z" fill="#1759BC"></path>
            </svg>
          </ic-button>
        </div>
      </nav>
      <div class="menu-status-version-container">
        <div class="menu-status">
          <ic-typography aria-label="app tag" class="menu-status-text" variant="label-uppercase">
            Beta
          </ic-typography>
        </div>
        <div class="menu-version">
          <ic-typography aria-label="app version" class="menu-version-text" variant="label">
            v1.0.0
          </ic-typography>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-navigation-menu>
`;

exports[`ic-navigation-menu should test slotted buttons: renders-with-slotted-buttons 1`] = `
<ic-navigation-menu status="Beta" version="v1.0.0">
  <mock:shadow-root>
    <div class="popout-modal"></div>
    <div aria-label="App menu" aria-modal="true" class="popout-menu" role="dialog">
      <span aria-hidden="true" class="navigation-landmark-text" id="navigation-landmark-text">
        Main navigation
      </span>
      <nav aria-hidden="true" aria-labelledby="navigation-landmark-text">
        <div class="menu-close-button-container">
          <ic-button class="button-size-large button-variant-icon menu-close-button" exportparts="button" id="menu-close-button">
            <mock:shadow-root>
              <ic-tooltip label="Close app menu" placement="bottom" target="ic-button-with-tooltip-menu-close-button">
                <button aria-label="Close app menu" class="button" id="ic-button-with-tooltip-menu-close-button" part="button" type="button">
                  <div class="icon-container">
                    <slot name="icon"></slot>
                  </div>
                  <slot></slot>
                </button>
              </ic-tooltip>
            </mock:shadow-root>
            <svg fill="none" height="14" slot="icon" viewBox="0 0 14 14" width="14" xmlns="http://www.w3.org/2000/svg">
              <path d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z" fill="#1759BC"></path>
            </svg>
          </ic-button>
        </div>
      </nav>
      <div class="menu-buttons-container">
        <slot name="buttons"></slot>
      </div>
      <div class="menu-status-version-container">
        <div class="menu-status">
          <ic-typography aria-label="app tag" class="menu-status-text" variant="label-uppercase">
            Beta
          </ic-typography>
        </div>
        <div class="menu-version">
          <ic-typography aria-label="app version" class="menu-version-text" variant="label">
            v1.0.0
          </ic-typography>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <ic-navigation-button label="button1" onclick="alert('test')" slot="buttons">
    <mock:shadow-root>
      <ic-button class="button-size-large button-variant-icon light" exportparts="button">
        <mock:shadow-root>
          <ic-tooltip label="button1" placement="bottom" target="ic-button-with-tooltip-">
            <button aria-label="button1" class="button" id="ic-button-with-tooltip-" part="button" type="button">
              <div class="icon-container">
                <slot name="icon"></slot>
              </div>
              <slot></slot>
            </button>
          </ic-tooltip>
        </mock:shadow-root>
        <slot name="icon" slot="icon"></slot>
      </ic-button>
    </mock:shadow-root>
    <svg fill="#000000" height="24px" slot="icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
      <path d="M0 0h24v24H0V0z" fill="none"></path>
      <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"></path>
    </svg>
  </ic-navigation-button>
</ic-navigation-menu>
`;

exports[`ic-navigation-menu should test slotted link: renders-with-slotted-link 1`] = `
<ic-navigation-menu status="Beta" version="v1.0.0">
  <mock:shadow-root>
    <div class="popout-modal"></div>
    <div aria-label="Navigation menu" aria-modal="true" class="popout-menu" role="dialog">
      <span aria-hidden="true" class="navigation-landmark-text" id="navigation-landmark-text">
        Main navigation
      </span>
      <nav aria-hidden="false" aria-labelledby="navigation-landmark-text">
        <div class="menu-close-button-container">
          <ic-button aria-label="Close navigation menu" class="menu-close-button" id="menu-close-button" size="large" variant="icon">
            <svg fill="none" height="14" slot="icon" viewBox="0 0 14 14" width="14" xmlns="http://www.w3.org/2000/svg">
              <path d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z" fill="#1759BC"></path>
            </svg>
          </ic-button>
        </div>
        <slot name="navigation"></slot>
      </nav>
      <div class="menu-status-version-container">
        <div class="menu-status">
          <ic-typography aria-label="app tag" class="menu-status-text" variant="label-uppercase">
            Beta
          </ic-typography>
        </div>
        <div class="menu-version">
          <ic-typography aria-label="app version" class="menu-version-text" variant="label">
            v1.0.0
          </ic-typography>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <a href="/" slot="navigation"></a>
</ic-navigation-menu>
`;

exports[`ic-navigation-menu should test slotted navigation and buttons: renders-with-slotted-navigation-and-buttons 1`] = `
<ic-navigation-menu status="Beta" version="v1.0.0">
  <mock:shadow-root>
    <div class="popout-modal"></div>
    <div aria-label="Navigation menu" aria-modal="true" class="popout-menu" role="dialog">
      <span aria-hidden="true" class="navigation-landmark-text" id="navigation-landmark-text">
        Main navigation
      </span>
      <nav aria-hidden="false" aria-labelledby="navigation-landmark-text">
        <div class="menu-close-button-container">
          <ic-button class="button-size-large button-variant-icon menu-close-button" exportparts="button" id="menu-close-button">
            <mock:shadow-root>
              <ic-tooltip label="Close navigation menu" placement="bottom" target="ic-button-with-tooltip-menu-close-button">
                <button aria-label="Close navigation menu" class="button" id="ic-button-with-tooltip-menu-close-button" part="button" type="button">
                  <div class="icon-container">
                    <slot name="icon"></slot>
                  </div>
                  <slot></slot>
                </button>
              </ic-tooltip>
            </mock:shadow-root>
            <svg fill="none" height="14" slot="icon" viewBox="0 0 14 14" width="14" xmlns="http://www.w3.org/2000/svg">
              <path d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z" fill="#1759BC"></path>
            </svg>
          </ic-button>
        </div>
        <slot name="navigation"></slot>
      </nav>
      <div class="menu-buttons-container menu-buttons-container-nav-item-above">
        <slot name="buttons"></slot>
      </div>
      <div class="menu-status-version-container">
        <div class="menu-status">
          <ic-typography aria-label="app tag" class="menu-status-text" variant="label-uppercase">
            Beta
          </ic-typography>
        </div>
        <div class="menu-version">
          <ic-typography aria-label="app version" class="menu-version-text" variant="label">
            v1.0.0
          </ic-typography>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <ic-navigation-button label="button1" onclick="alert('test')" slot="buttons">
    <mock:shadow-root>
      <ic-button class="button-size-large button-variant-icon light" exportparts="button">
        <mock:shadow-root>
          <ic-tooltip label="button1" placement="bottom" target="ic-button-with-tooltip-">
            <button aria-label="button1" class="button" id="ic-button-with-tooltip-" part="button" type="button">
              <div class="icon-container">
                <slot name="icon"></slot>
              </div>
              <slot></slot>
            </button>
          </ic-tooltip>
        </mock:shadow-root>
        <slot name="icon" slot="icon"></slot>
      </ic-button>
    </mock:shadow-root>
    <svg fill="#000000" height="24px" slot="icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
      <path d="M0 0h24v24H0V0z" fill="none"></path>
      <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"></path>
    </svg>
  </ic-navigation-button>
  <ic-navigation-item class="light navigation-item" href="/" label="Navigation 1" role="listitem" slot="navigation">
    <mock:shadow-root>
      <ic-tooltip class="tooltip-navigation-item" label="Navigation 1" placement="right" target="navigation-item">
        <a class="link" href="/" part="link">
          <ic-typography variant="label">
            Navigation 1
          </ic-typography>
        </a>
      </ic-tooltip>
    </mock:shadow-root>
  </ic-navigation-item>
</ic-navigation-menu>
`;

exports[`ic-navigation-menu should test slotted navigation group: renders-with-slotted-navigation-group 1`] = `
<ic-navigation-menu status="Beta" version="v1.0.0">
  <mock:shadow-root>
    <div class="popout-modal"></div>
    <div aria-label="Navigation menu" aria-modal="true" class="popout-menu" role="dialog">
      <span aria-hidden="true" class="navigation-landmark-text" id="navigation-landmark-text">
        Main navigation
      </span>
      <nav aria-hidden="false" aria-labelledby="navigation-landmark-text">
        <div class="menu-close-button-container nav-group-first">
          <ic-button aria-label="Close navigation menu" class="menu-close-button" id="menu-close-button" size="large" variant="icon">
            <svg fill="none" height="14" slot="icon" viewBox="0 0 14 14" width="14" xmlns="http://www.w3.org/2000/svg">
              <path d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z" fill="#1759BC"></path>
            </svg>
          </ic-button>
        </div>
        <slot name="navigation"></slot>
      </nav>
      <div class="menu-status-version-container">
        <div class="menu-status">
          <ic-typography aria-label="app tag" class="menu-status-text" variant="label-uppercase">
            Beta
          </ic-typography>
        </div>
        <div class="menu-version">
          <ic-typography aria-label="app version" class="menu-version-text" variant="label">
            v1.0.0
          </ic-typography>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <ic-navigation-group class="expanded" expandable="true" label="Navigation group" role="listitem" slot="navigation">
    <mock:shadow-root>
      <button aria-expanded="true" aria-haspopup="false" class="light navigation-group" tabindex="0">
        <ic-typography variant="label">
          Navigation group
        </ic-typography>
      </button>
      <ul class="grouped-links-wrapper" style="--navigation-child-items-height: auto;">
        <slot></slot>
      </ul>
    </mock:shadow-root>
  </ic-navigation-group>
</ic-navigation-menu>
`;

exports[`ic-navigation-menu should test slotted navigation item: renders-with-slotted-navigation-item 1`] = `
<ic-navigation-menu status="Beta" version="v1.0.0">
  <mock:shadow-root>
    <div class="popout-modal"></div>
    <div aria-label="Navigation menu" aria-modal="true" class="popout-menu" role="dialog">
      <span aria-hidden="true" class="navigation-landmark-text" id="navigation-landmark-text">
        Main navigation
      </span>
      <nav aria-hidden="false" aria-labelledby="navigation-landmark-text">
        <div class="menu-close-button-container">
          <ic-button aria-label="Close navigation menu" class="menu-close-button" id="menu-close-button" size="large" variant="icon">
            <svg fill="none" height="14" slot="icon" viewBox="0 0 14 14" width="14" xmlns="http://www.w3.org/2000/svg">
              <path d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z" fill="#1759BC"></path>
            </svg>
          </ic-button>
        </div>
        <slot name="navigation"></slot>
      </nav>
      <div class="menu-status-version-container">
        <div class="menu-status">
          <ic-typography aria-label="app tag" class="menu-status-text" variant="label-uppercase">
            Beta
          </ic-typography>
        </div>
        <div class="menu-version">
          <ic-typography aria-label="app version" class="menu-version-text" variant="label">
            v1.0.0
          </ic-typography>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <ic-navigation-item class="light navigation-item" href="/" label="Navigation 1" role="listitem" slot="navigation">
    <mock:shadow-root>
      <ic-tooltip class="tooltip-navigation-item" label="Navigation 1" placement="right" target="navigation-item">
        <a class="link" href="/" part="link">
          <ic-typography variant="label">
            Navigation 1
          </ic-typography>
        </a>
      </ic-tooltip>
    </mock:shadow-root>
  </ic-navigation-item>
</ic-navigation-menu>
`;
